モノレポで管理できるようにする

himanoa opened the pull request:
cloudflareのランタイムをサポートしようとしたが、bigintを使っているため同じコードベースで動かすことは困難だった。

そのため、既存の実装とcloudflare workers用のパッケージを分離する必要がある。

このPRではパッケージを分離してもおなじリポジトリで管理できるようにするためyarn workspaceで既存のコードを管理できるようにする

himanoa committed:
Initialize js-scrambler package

himanoa committed:
Migrated package.json in js-scrambler package

- Add support ESModule

himanoa committed:
Enable yarn workspace

- Organized .gitignore

himanoa committed:
Move implementation and test from root package to js-scrambler package

himanoa committed:
Fix a test workflow where doesn't work

uneco commented on packages/js-scrambler/package.json:
```diff
@@ -0,0 +1,41 @@
+  "keywords": [
+    "Scrambler"
+  ],
+  "author": "AOKI Yuto <yuto.aoki@japan-d2.com> (https://japan-d2.com)",
```
😮‍💨

naporin0624 commented on package.json:
```diff
@@ -1,38 +1,11 @@
   "version": "0.0.5",
   "description": "Converts an integer to another integer that is unpredictable and reversible.",
   "main": "lib/index.js",
   "typings": "lib/index.d.ts",
```
些細だけどいらんくね？

uneco commented on packages/js-scrambler/package.json:
```diff
@@ -0,0 +1,41 @@
+  "name": "@kanamone/js-scrambler",
+  "version": "0.0.5",
+  "repository": "https://github.com/kanamone/js-scrambler",
+  "author": "AOKI Yuto <aoki@u-ne.co>",
```
😮‍💨

uneco commented on packages/js-scrambler/package.json:
```diff
@@ -0,0 +1,41 @@
+    "Scrambler"
+  ],
+  "author": "AOKI Yuto <yuto.aoki@japan-d2.com> (https://japan-d2.com)",
+  "homepage": "https://github.com/japan-d2/js-scrambler",
```
😮‍💨

uneco reviewed (changes requested):
😮‍💨

naporin0624 commented on packages/js-scrambler/package.json:
```diff
@@ -0,0 +1,41 @@
+  "author": "AOKI Yuto <aoki@u-ne.co>",
+  "license": "MIT",
+  "type": "module",
+  "main": "lib/index.js",
+  "types": "lib/index.d.ts",
```
せっかくだしこの際に dual package 対応するとよさそう `publint` と `tsup` でいちころっぽい

himanoa replied:
これはやるかあ

euxn23 commented on .gitignore:
```diff
@@ -1,110 +1,2 @@
-
-# lib
-/lib/
+packages/**/lib
```
深い階層の `lib` が消えかねないので `*` の方が安全かも

himanoa replied:
libディレクトリ自体は存在しなかったらバンドラが作ってくれるだろうし、initialでなくていいかなーとおもってこの指定にしましたがまあどっちでもいいんで直しときます

himanoa committed:
Fix gitignore

** to *

Co-authored-by: euxn23 <euxn23@gmail.com>

himanoa committed:
Setup publint and tsup

- Change build output directory name from lib to path.
- Fix a publint error where not found exports field
- Add support dual-package

Co-authored-by: naporin0624 <naporin0624@gmail.com>

himanoa committed:
Move to tsconfig.json and eslintrc from root to js-scrambler package

himanoa committed:
Fix a author field and import-path

Co-authored-by: uneco <aoki@u-ne.co>

himanoa committed:
Add ci-tasks where publint and lint

himanoa commented:
## 確認したこと

- ripgrep japan-d2 がヒットしないこと
- 指摘内容が直っていること

naporin0624 commented on yarn.lock:
pnpm にしたい
